### ISSUE TITLE
Incorrect comment on ThreadLocal for avoiding nullness warnings

### ISSUE THREAD

**Issue by jpschewe** [typetools/checker-framework#4708]
According to https://github.com/typetools/jdk/blob/master/src/java.base/share/classes/java/lang/ThreadLocal.java#L84-L89 I should be able to add a warning suppression to specify a non-null type for ThreadLocal

```java
package net.mtu.eggplant.checker;

import java.text.NumberFormat;

@SuppressWarnings("nullness:type.argument.type.incompatible") // initialValue returns non-null
public final class NonNullThreadLocal extends ThreadLocal<NumberFormat> {
  @Override
  protected NumberFormat initialValue() {
    final NumberFormat format = NumberFormat.getInstance();
    // setup the number format instance to be 2 decimal places
    format.setMaximumFractionDigits(2);
    format.setMinimumFractionDigits(2);
    return format;
  }
}
```

gives the following warning
```
/home/jpschewe/projects/checker-bugs/src/main/java/net/mtu/eggplant/checker/NonNullThreadLocal.java:6: error: [type.argument] incompatible type argument for type parameter T of ThreadLocal.
public final class NonNullThreadLocal extends ThreadLocal<NumberFormat> {
                                                          ^
  found   : @Initialized @NonNull NumberFormat
  required: [extends @Initialized @Nullable Object super null (NullType)]
```

After some more experimentation with this I see that the correct annotation is "@SuppressWarnings("nullness:type.argument")", so it seems that the documentation just needs to be updated.

I see that https://github.com/typetools/checker-framework/issues/1572 is open for a more permanent solution to this problem. That seems like a good thing. Although I would argue that the real problem is that ThreadLocal isn't an abstract class.
